src/foraging_gui/Foraging.py

```diff
@@ -109,6 +110,21 @@ def __init__(self, parent=None,box_number=1,start_bonsai_ide=True):
         # Load User interface
         self._LoadUI()
 
+        # create AINDBehaviorSession model to be used and referenced for session info
+        self.behavior_session_model = AindBehaviorSessionModel(
+            experiment=self.Task.currentText(),
+            experimenter=[self.Experimenter.text()],
+            date=datetime.now(),   # update when folders are created
+            root_path='',         # update when created
+            session_name= '',   # update when date and subject are filled in
```

Comment:
I would pass a null for the session name since this field is generated automatically at model instantiation.

Reply:
The "AindBehaviorSessionModel" generates the name? Even if you pass in a name?

The docs say "Name of the session. This will be used to create a folder in the root path." (https://allenneuraldynamics.github.io/Aind.Behavior.Services/api.base/session.html)
Does creating the schema make the folder?

Right now our code sets the session name and makes a folder when the session starts. Note that the user can open the GUI and do things before starting the session. I think Micah was planning on setting the session name to the empty string until the session actually starts, and then pass in the final session name

Reply:
I was playing around with that but I think the time format is different that the one currently used. It seems like the  [format_datetime ](https://github.com/AllenNeuralDynamics/Aind.Behavior.Services/blob/0654486951b766ea6ef2f767c75949adf25efa1b/src/aind_behavior_services/utils.py#L411C1-L420C1) uses "%Y-%m-%dT%H%M%SZ" and we use "%Y-%m-%d_%H-%M-%S".  Is there a way to format the date differently? Or @alexpiet could we change the date format on our side?

Reply:
When did that become the time standard ? I know last year we had a LOT of discussion about time format and settled on "%Y-%m-%d_%H-%M-%S"

Maybe the use case is to override the validator until the session starts and we have a final session name, and then allow the validator to work.

Reply:
It looks like the other pipelines are all using "%Y-%m-%d_%H-%M-%S" based on code ocean data assets.

Reply:
I dont think it has become a standard, hence the "proposed". If the current standard is this one: https://aind-data-schema.readthedocs.io/en/latest/data_organization.html. When I checked this I found it odd that the same document uses at least 4 different standards (some of which are in direct conflict with what is written in the document itself, like not being timezone aware).

Additionally, the string does not seem to follow the ISO8061 (or at least not the native python implementation). Such that the following example fails. I would strongly encourage people to standardize formats that allow us to use already existing tools.

```python
import datetime

tz = "2022-04-26T114809"
tz_alex = "2022-04-26_11-48-09"

print(datetime.datetime.fromisoformat(tz))
# 2022-04-26 11:48:09
print(datetime.datetime.fromisoformat(tz_alex))
# Throws error ValueError: Invalid isoformat string: '2022-04-26_11-48-09'
```

Reply:
I see, well, I think we should keep using the existing standard for consistency with existing data, other pipelines, and AIND metadata. 

(personal aside, I think 11-48-09 is much easier to read than T114809)